app/models/ms_media_files.php

```diff
@@ -772,4 +790,271 @@ public function formatPublishedText($pn_published=null) {
  		return $vs_publish_text;
 	}
 	# ------------------------------------------------------
+ 	# Persistent identifier (DOI, ARK) methods
+ 	# ------------------------------------------------------
+ 	/*
+ 	 *
+ 	 */
+ 	public function getARK($va_reserved=true, $vs_user_fname=null, $vs_user_lname=null) {
+ 		$va_ark = array();
+ 		if(!$this->getPrimaryKey()) { 
+ 			$va_ark["success"] = false;
+ 			$va_ark["error"] = "Media file does not have media file id.";
+ 		}
+ 		if($this->get("ark")) {
+ 			$va_ark["success"] = false;
+ 			$va_ark["error"] = "Media file already has an ARK.";
+ 		}
+
+ 		if ($va_reserved) {
+ 			$va_metadata = array("_status" => "reserved", "_export" => "no");
+ 		} else {
+ 			if (!vs_user_fname || !vs_user_lname) {
+ 				$va_ark["success"] = false;
+ 				$va_ark["error"] = "User first and last names not provided.";
+ 			}
+ 			
+ 			$vs_mime_type = strtolower(
+ 				$this->getMediaInfo('media', 'original', 'MIMETYPE'));
+			$va_mime_type = explode('/', $vs_mime_type);
+	
+			switch($va_mime_type[0]) {
+				case 'image':
+					$vs_resource_type = 'Image';
+					break;
+				case 'audio':
+				case 'video':
+					$vs_resource_type = 'Audiovisual';
+					break;
+				case 'application':
+					switch($vs_mime_type) {
+						case 'application/ply':
+						case 'application/stl':
+						case 'application/surf':
+						case 'text/prs.wavefront-obj':
+							$vs_resource_type = 'Model';	
+							break(2);
+						case 'application/zip':
+							$vs_resource_type = 'Dataset';	
+							break(2);
+						case 'application/dicom':
+							$vs_resource_type = 'Image';	
+							break(2);
+					}
+				default:
+					$vs_resource_type = 'Other';
+					break;
+			}
+
+			$va_metadata = array(
+				"datacite.creator" => trim($vs_user_fname.' '.$vs_user_lname),
+				"datacite.title" => 
+					"M".$this->get("media_id")."-".$this->getPrimaryKey(),
+				"datacite.publisher" => "MorphoSource.org",
+				"datacite.publicationyear" => date("Y"),
+				"datacite.resourcetype" => $vs_resource_type,	
+				"_target" => "http://MorphoSource.org/index.php/Detail/".
+					"MediaDetail/Show/media_file_id/".$this->getPrimaryKey(),
+				"_status" => "public",
+				"_export" => "yes",
+				"_profile" => "datacite"
+			);
+ 		}
+
+ 		$o_ark = new ARK();
+		try {
+			if ($vs_ark = $o_ark->createARK("M".$this->getPrimaryKey(), 
+				$va_metadata)) 
+			{
+				# --- record the ARK in the DB
+				$this->set("ark", $vs_ark);
+				if ($va_reserved) { $this->set("ark_reserved", 1); }
+				$this->setMode(ACCESS_WRITE);
+				$this->update();
+				$va_ark["success"] = true;
+			} else { 
+				$va_ark["success"] = false;
+				$va_ark["error"] = "Could not get ARK for media: ".
+					$o_ark->getError();
+			}	
+		} catch (Exception $e) {
+			$va_ark["success"] = false;
+			$va_ark["error"] = "Could not get ARK for media: ".$e->getMessage();
+		}
+		return $va_ark;
+ 	}
+ 	/*
```

Comment:
if I understand correctly, createARK will return false if there is a problem (e.g. server timeout, error).  in that case $vs_ark will be set to false, correct?  Just want to make sure if it is set to false, it will be expected and being handled.

Reply:
Yeah, that is how it works. If createARK fails it will return false and that will be handled by setting $vs_ark["success"] to false and getting an error message for $vs_ark["error"]